api/client/samples/analogous_years/lib/final_ranks_computation.py

```diff
@@ -78,17 +78,16 @@ def time_series(client, data_series, initial_date, final_date):
     :return: A dataframe with data from the relevant dates
     """
     logger = client.get_logger()
-    entities = {'metric_id', 'item_id', 'region_id', 'source_id', 'frequency_id', 'start_date'}
```

Comment:
haha this proved my point earlier. just replacing every occurrence of `start_date` by `start_date_bound` is a potential source of bugs

Reply:
`haha` :-o

Reply:
You are right, I started this PR. with the idea that the organic `start_date` (that comes with the data series itself) should be separated from the start_date bound human or the program generated. Because, I wanted to use that. So, I thought it might become a readability issue. Hence, I replaced all the occurrences of `start_date` which came from it being a bound at the beginning of the program. This is the only place that I remember that had to be replaced which I did not foresee. Now, I have not used the organic `start_date` at all. So, I can potentially reverse all the changes. However, I want to keep in case in the future (or even in this PR) we end up using it.

Reply:
Discussed offline. Reversed some of the naming changes and restricted most of the changes to one module for the sake of brevity.